### PR TITLE
HDDS-7440. Delete package-info class for org.apache.hadoop.io.erasurecode.rawcoder to avoid conflict with Hadoop

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Trying to use Ozone FS jar in Spark points out conflict with one of Hadoop's packages:

```
scala> :require /opt/ozone/share/ozone/lib/ozone-filesystem-hadoop3-1.3.0-SNAPSHOT.jar
The path '/opt/ozone/share/ozone/lib/ozone-filesystem-hadoop3-1.3.0-SNAPSHOT.jar' cannot be loaded, it contains a classfile that already exists on the classpath: org.apache.hadoop.io.erasurecode.rawcoder.package-info
```

https://issues.apache.org/jira/browse/HDDS-7440

## How was this patch tested?

Regular CI:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/3368073737

After the update Spark still complains, but only about non-shaded third-party classes.  Those will be addressed in separate task.

### CODE REVIEW

File: hadoop-hdds/dev-support/checkstyle/suppressions.xml [hunk 1]
```diff
@@ -18,4 +18,5 @@
 
 <suppressions>
   <suppress checks="JavadocPackage" files="[\\/]src[\\/]test[\\/].*"/>
+  <suppress checks="JavadocPackage" files="src/main/java/org/apache/hadoop/io/erasurecode/rawcoder"/>
```
Comment (myskov) [inline hadoop-hdds/dev-support/checkstyle/suppressions.xml:21]: BTW what about dropping `JavadocPackage` rule? I believe it's adopted from Hadoop code style, but it doesn't seem to make much sense.

Comment (myskov) [reply hadoop-hdds/dev-support/checkstyle/suppressions.xml:21]: Removing the rule will allow to remove useless (in most cases) package-info files.

**Comment by kerneltime** [comment]
@adoroszlai would shading be a better approach as we might not want EC package to be picked by the application but be based on the actual testing done in the released version?

**Comment by adoroszlai** [comment]
> would shading be a better approach?

I don't think shading helps here.  We need this Hadoop-specific package name to access some protected members in some Hadoop classes in the same package.  The only real code in this package is the utility to access the Hadoop code.  If we shade the Ozone's classes in this package then I don't think it will be able to access Hadoop classes.  If we shade both Hadoop and Ozone, we're back to duplicate package-info class.

The other possible solution is to copy all related code from Hadoop, but I think the goal in this case was to avoid that.

**Comment by adoroszlai** [comment]
Thanks @kerneltime, @myskov, @sodonnel for the review.

**Pull request merged**